utils.py

```diff
@@ -0,0 +1,160 @@
+# General imports.
+import os
+import zipfile
+from glob import glob
+from shutil import copyfile
+from shutil import copytree
+from shutil import copyfileobj
+from shutil import move
+import h5py
+import numpy as np
+from skimage.measure import regionprops
+import requests
+from tqdm import tqdm
+# saving images to tif formats
+import imageio.v3 as imageio
+
+# this is the folder where all data is stored.
+data_folder = "data"
+os.makedirs(data_folder, exist_ok=True)
+
+# the data is on zenodo
+data_url = "https://zenodo.org/record/5092850/files/covid-if-groundtruth.zip?download=1"
+
+# download the data using the requests library
+
+# this function is for downloading the data.
+# you don't need to understand what's going on here.
+def download_url(url, path):
+    if os.path.isfile(path):
+        return 
+    with requests.get(url, stream=True) as r:
+        if r.status_code != 200:
+            r.raise_for_status()
+            raise RuntimeError(f"Request to {url} returned status code {r.status_code}")
+        file_size = int(r.headers.get("Content-Length", 0))
+        desc = f"Download {url} to {path}"
+        if file_size == 0:
+            desc += " (unknown file size)"
+        with tqdm.wrapattr(r.raw, "read", total=file_size, desc=desc) as r_raw, open(path, "wb") as f:
+            copyfileobj(r_raw, f)
+
+
+# run the download function to donwnload the data to the file 'data.zip' in the data folder
+download_url(data_url, os.path.join(data_folder, "data.zip"))
+
+# unzip the data using the zipfile library
+# function for unzipping the archive we have just downloaded and then removing the zip
+def unzip(zip_path, dst, remove=True):
+    with zipfile.ZipFile(zip_path, "r") as f:
+        f.extractall(dst)
+    if remove:
+        os.remove(zip_path)
+unzip(os.path.join(data_folder, "data.zip"), data_folder, remove=False)
```

Comment:
See above. (Same for all the global function calls and variable definitions below.)